common/djangoapps/third_party_auth/tests/test_provider.py

```diff
@@ -114,45 +116,79 @@ def test_providers_displayed_for_login(self):
         provider_ids = [
             idp.provider_id
             for idp in provider.Registry.displayed_for_login(tpa_hint=hidden_provider.provider_id)
         ]
         assert hidden_provider.provider_id in provider_ids
 
-        # New providers are hidden (ie, not flagged as 'visible') by default
-        # The tpa_hint parameter should work for these providers as well
+    def test_tpa_hint_hidden_provider_displayed_for_login(self):
+        """
+        Tests to ensure that an implicitly enabled-but-not-visible provider is presented
+        for use in the UI when the "tpa_hint" parameter is specified.
+        New providers are hidden (ie, not flagged as 'visible') by default
+        The tpa_hint parameter should work for these providers as well.
+        """
+
         implicitly_hidden_provider = self.configure_linkedin_provider(enabled=True)
         provider_ids = [
             idp.provider_id
             for idp in provider.Registry.displayed_for_login(tpa_hint=implicitly_hidden_provider.provider_id)
         ]
         assert implicitly_hidden_provider.provider_id in provider_ids
 
-        # Disabled providers should not be matched in tpa_hint scenarios
+    def test_tpa_hint_disabled_hidden_provider_displayed_for_login(self):
+        """
+        Disabled providers should not be matched in tpa_hint scenarios
+        """
         disabled_provider = self.configure_twitter_provider(visible=True, enabled=False)
         provider_ids = [
             idp.provider_id
             for idp in provider.Registry.displayed_for_login(tpa_hint=disabled_provider.provider_id)
         ]
         assert disabled_provider.provider_id not in provider_ids
 
-        # Providers not utilized for learner authentication should not match tpa_hint
+    def test_tpa_hint_no_log_hidden_provider_displayed_for_login(self):
+        """
+        Providers not utilized for learner authentication should not match tpa_hint
+        """
         no_log_in_provider = self.configure_lti_provider()
         provider_ids = [
             idp.provider_id
             for idp in provider.Registry.displayed_for_login(tpa_hint=no_log_in_provider.provider_id)
         ]
         assert no_log_in_provider.provider_id not in provider_ids
 
+    @with_site_configuration(SITE_DOMAIN_A)
+    def test_get_current_site_oauth_provider(self):
+        """
+        Verify that correct provider for current site is returned even if same backend is used for multiple sites.
+        """
+        current_provider = self.configure_google_provider(visible=True, enabled=True, site=Site.objects.get_current())
+        assert current_provider.enabled_for_current_site is True
+        site_b = Site.objects.get_or_create(domain=SITE_DOMAIN_B, name=SITE_DOMAIN_B)[0]
+        # Create google provider but for other site
+        other_site_provider = self.configure_google_provider(visible=True, enabled=True, site=site_b)
+        provider_ids = provider.Registry.displayed_for_login()
```

Comment:
@navinkarkera: Thanks so much for the test. I've got a couple of minor suggestions that might have helped me understand this test a bit more quickly:
1. Consider moving `with_site_configuration(SITE_DOMAIN_A)` inside the test using a parallel `with` to the second-half of the test.
2. Add an assertion that `Site.objects.get_current()` is `SITE_DOMAIN_A` near the top. This might be enough, without moving the decorator in the test, to have made this click a bit more.
3. Can you add a blank-line and comment before the `provider.Registry.displayed_for_login()` line? I presume this works with the current site, but it took me a little time to understand this.

Anyway - some minor changes to consider that would have helped someone like me who was only looking at this single test.
5. T

Reply:
@robrap Thanks for the suggestions! Updated the test so that it is easier to read and understand.